### PR TITLE
Use GitHub Actions (`gha`) cache for Docker builds

### PR DESCRIPTION
**Pull request by siggy**
Docker introduced a `gha` cache type, removing the need to manage our
own Docker cache in CI.

Modify the `docker-build` action to set required `ACTIONS_CACHE_URL` and
`ACTIONS_RUNTIME_TOKEN` environment variables, consumed by
`docker buildx` commands.

Cached Docker build times reduced to 15s ~ 1m45s.

Screenshot 2023-11-11 at 11 26 23 PM

### CODE REVIEW

File: bin/_docker.sh [hunk 1]
```diff
@@ -69,6 +72,8 @@ docker_build() {
 
     if [ "$DOCKER_BUILDKIT_CACHE" ]; then
       cache_params="--cache-from type=local,src=${DOCKER_BUILDKIT_CACHE} --cache-to type=local,dest=${DOCKER_BUILDKIT_CACHE},mode=max"
+    elif [ "$ACTIONS_CACHE_URL" ]; then
```
Comment (mateiidavid) [inline bin/_docker.sh:75]: The condition above (`$DOCKER_BUILDKIT_CACHE`) might be fine to clean? Doing a grep through the repo, I can only find references to it in:
* The build action (which has been cleaned up)
* This script
* The prune script (which we no longer call)

...so it miiiiight be safe to rm?

```
:; rg 'DOCKER_BUILDKIT_CACHE'
bin/docker-cache-prune
8:export DOCKER_BUILDKIT_CACHE=${DOCKER_BUILDKIT_CACHE:-}
10:manifest_sha=$(jq -r .manifests[0].digest < "${DOCKER_BUILDKIT_CACHE}/index.json")
13:while IFS= read -r line; do files+=("$line"); done <<< "$(jq -r '.manifests[].digest | sub("^sha256:"; "")' < "${DOCKER_BUILDKIT_CACHE}/blobs/sha256/$manifest")"
14:for file in "${DOCKER_BUILDKIT_CACHE}"/blobs/sha256/*; do

bin/_docker.sh
19:export DOCKER_BUILDKIT_CACHE=${DOCKER_BUILDKIT_CACHE:-}
70:    if [ "$DOCKER_BUILDKIT_CACHE" ]; then
71:      cache_params="--cache-from type=local,src=${DOCKER_BUILDKIT_CACHE} --cache-to type=local,dest=${DOCKER_BUILDKIT_CACHE},mode=max"
```

File: .github/actions/docker-build/action.yml [hunk 2]
```diff
@@ -62,10 +55,6 @@ runs:
         echo "${image_digest}"
         echo "DIGEST=${{ inputs.docker-registry }}/${{ inputs.component }}@${image_digest}" >> "$GITHUB_OUTPUT"
 
-    # Delete older docker images to prevent cache bloat.
-    - run: bin/docker-cache-prune
```
Comment (mateiidavid) [inline .github/actions/docker-build/action.yml:66]: Could we also remove the script from the file tree? I don't think we're using it anywhere else.

```
:; rg 'docker-cache-prune'
actions/docker-build/action.yml
66:    - run: bin/docker-cache-prune
```

Can also be done as a follow-up. It's always dangerous when removing (seemingly) unused scripts.

**Review by mateiidavid — APPROVED** [review]
This is really cool! Thanks a lot! 🚢

File: .github/actions/docker-build/action.yml (hunk 2 above)
Comment (siggy) [reply .github/actions/docker-build/action.yml:66]: i did consider removing this script, along with references to `DOCKER_BUILDKIT_CACHE`, but i wasn't sure if anyone was using these for their own local development. happy to remove all this if not the case.

File: bin/_docker.sh (hunk 1 above)
Comment (siggy) [reply bin/_docker.sh:75]: yup happy to remove, relates to my other comment.

**Review by adleong — APPROVED** [review]

**Review by olix0r — APPROVED** [review]

**Pull request merged**